src/froidure-pin-base.cpp

```diff
@@ -0,0 +1,725 @@
+//
+// libsemigroups_pybind11
+// Copyright (C) 2024 James Mitchell
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+// C std headers....
+// TODO complete or delete
+
+// C++ stl headers....
+// TODO complete or delete
+
+// libsemigroups headers
+#include <libsemigroups/froidure-pin-base.hpp>
+
+// pybind11....
+// #include <pybind11/chrono.h>
+// #include <pybind11/functional.h>
+#include <pybind11/pybind11.h>
+// #include <pybind11/stl.h>
+// TODO uncomment/delete
+
+// libsemigroups_pybind11....
+#include "main.hpp"  // for init_froidure_pin_base
+
+namespace py = pybind11;
+
+namespace libsemigroups {
+
+  void init_froidure_pin_base(py::module& m) {
+    py::class_<FroidurePinBase> thing(m,
+                                      "FroidurePinBase",
+                                      R"pbdoc(
+Base class for :any:`FroidurePin` containing non-element specific data and
+member functions.
+
+.. seealso::  :any:`FroidurePin`.)pbdoc");
```

Comment:
Is it worth mentioning here that it's an abstract base class, or at least saying you can't construct a `FroidurePinBase`?

Reply:
Yeah almost certainly. Will do later in the afternoon, thanks for the review! @Joseph-Edwards